### DISCUSSION TITLE
Problems self-hosting on Debian (from Issue #3062)

### DISCUSSION THREAD

**Discussion by mngshm** [ente-io/ente#3901] [Q&A]
@jerroder I am answering all your queries here for now. For those who might feel contextless, please check issue #3062. 

Hey, extremely sorry for not making the documentation verbose. That documentation is quite old now and I made a updated [version](https://help.ente.io/self-hosting/guides/standalone-ente) recently. But I think it still does lack some verbosity, I'll make sure to add more instructions on my next working day.  

>Now in Start museum, I need to set the variable ENTE_DB_USER to postgres but if I use air for live reloads, the user is ente_user?

Extremely sorry for the above. It looks like this was mistake. Just now I ran Air (for the first time) and you're right it has to be `ENTE_DB_USER=postgres` both the times. But, if you've already exported it as postgres then you don't have to do the variable setting before running air (again, sorry I'll improve the doc).

>When I ran this, it downloaded a bunch of stuff and then exited without any error. Is this expected

The reason we mentioned the command `go run cmd/museum/main.go` to be run from `ente/server` dir is because the `configurations/local.yaml` and the whole environment is in the root directory of museum (server). Maybe try cleaning your go cache by running `go clean -modcache` and re-running the command from the servers root dir which is `ente/server`. please let me know if its not fixed. 

Also, that isn't the expected behavior, it should be something like the below screenshot (from my own server). 

![Capture-Nov-02-00-11-19](https://github.com/user-attachments/assets/0aa1ce55-6266-4a45-8c63-ffb084262f61)
_Please note I'm running museum with Air_

>Do I need to fill museum.yaml with DB user and password?

Yes.

>And museum.yaml should be created "in the current directory" but where is that? ente/? ente/server/?

It should be in the `ente/server` directory, but this is only comes in use if you want to override stuff which is there in `ente/server/configurations/local.yaml`. 

>If it works eventually, how do I stop, start and reload it?

What exactly ? Museum or Air ? 

For both Museum and Air, you can do a simple ctrl+c and make sure to kill the port `:8080` (or whatever you have custom configured), so that the next time you can start museum without any errors.   

Regarding reloads in Air, I'm not sure if Air has any reload kind of feature. 

You can do a simple `lsof -i :8080` and kill the PID.

**Comment by AlexisEvo** [comment]
Sorry to bump, but came across this from Google. If anyone else runs into the referenced error:

```
package command-line-arguments
        imports github.com/ente-io/museum/pkg/api
        imports github.com/ente-io/museum/pkg/controller
        imports github.com/ente-io/museum/pkg/controller/access
        imports github.com/ente-io/museum/pkg/repo
        imports github.com/ente-io/museum/pkg/utils/crypto
        imports github.com/GoKillers/libsodium-go/cryptogenerichash: build constraints exclude all Go files in /home/xxx/go/pkg/mod/github.com/!go!killers/libsodium-go@v0.0.0-20171022220152-dd733721c3cb/cryptogenerichash
package command-line-arguments
        imports github.com/ente-io/museum/pkg/api
        imports github.com/ente-io/museum/pkg/controller
        imports github.com/ente-io/museum/pkg/controller/access
        imports github.com/ente-io/museum/pkg/repo
        imports github.com/ente-io/museum/pkg/utils/crypto
        imports github.com/GoKillers/libsodium-go/cryptosecretbox: build constraints exclude all Go files in /home/xxx/go/pkg/mod/github.com/!go!killers/libsodium-go@v0.0.0-20171022220152-dd733721c3cb/cryptosecretbox
```
        
You need to install gcc.
